components/inputs/input-percent.js

```diff
@@ -0,0 +1,180 @@
+			maxFractionDigits: { type: Number, attribute: 'max-fraction-digits' },
+			/**
+			 * Minimum number of decimal values to show.
+			 */
+			minFractionDigits: { type: Number, attribute: 'min-fraction-digits' },
+			/**
+			 * Placeholder text
+			 */
+			placeholder: { type: String },
+			/**
+			 * Indicates that a value is required
+			 */
+			required: { type: Boolean },
+			/**
+			 * Text for additional screenreader and mouseover context
+			 */
+			title: { type: String },
+			/**
+			 * Value of the input
+			 */
+			value: { type: Number }
+		};
+	}
+
+	static get styles() {
+		return [ super.styles,
+			css`
+				:host {
+					--d2l-input-text-align: end;
+					display: inline-block;
+					position: relative;
+					width: 100%;
+				}
+				:host([hidden]) {
+					display: none;
+				}
+				:host([disabled]) [slot="right"] {
+					opacity: 0.5;
+				}
+				[slot="right"] {
+					box-sizing: border-box;
+					cursor: default;
+					padding-left: 0.2rem;
+					padding-right: 0.55rem;
+				}
+				:host([dir="rtl"]) [slot="right"] {
+					padding-left: 0.55rem;
+					padding-right: 0.2rem;
+				}
+			`
+		];
+	}
+
+	constructor() {
+		super();
+		this.autofocus = false;
+		this.disabled = false;
+		this.labelHidden = false;
+		this.required = false;
+	}
+
+	firstUpdated(changedProperties) {
+		super.firstUpdated(changedProperties);
+		if (!this.label) {
+			console.warn('d2l-input-percent component requires label text');
+		}
+	}
+
+	render() {
+		return html`
+			<d2l-input-number
+				autocomplete="${ifDefined(this.autocomplete)}"
+				?autofocus="${this.autofocus}"
+				@blur="${this._handleBlur}"
+				@change="${this._handleChange}"
+				?disabled="${this.disabled}"
+				.forceInvalid="${this.invalid}"
+				hide-invalid-icon
+				input-width="${ifDefined(this.inputWidth)}"
+				label="${ifDefined(this.label)}"
+				?label-hidden="${this.labelHidden}"
+				max="100"
+				max-fraction-digits="${ifDefined(this.maxFractionDigits)}"
+				min="0"
+				min-fraction-digits="${ifDefined(this.minFractionDigits)}"
+				name="${ifDefined(this.name)}"
+				?noValidate="${this.noValidate}"
+				placeholder="${ifDefined(this.placeholder)}"
+				?required="${this.required}"
+				?skeleton="${this.skeleton}"
+				title="${ifDefined(this.title)}"
+				value="${ifDefined(this.value)}">
+					<span slot="right" @click="${this._handleSymbolClick}">%</span>
+					<slot slot="after" name="after"></slot>
+			</d2l-input-number>
+		`;
+	}
+
+	focus() {
+		const elem = this.shadowRoot.querySelector('d2l-input-number');
+		if (elem) elem.focus();
+	}
+
+	async validate() {
+		const inputNumberElem = this.shadowRoot.querySelector('d2l-input-number');
+		await inputNumberElem.updateComplete;
+		const childErrors = await inputNumberElem.validate();
+		const errors = await super.validate();
+		return [...childErrors, ...errors];
+	}
+
+	async _handleChange(e) {
+		const oldValue = this.value;
+		this.value = e.target.value;
+
+		if (e.target.value === undefined) {
```

Comment:
The `target.value` will from `d2l-input-number` will be `undefined` if the user entered an invalid value, so we reset to the old value.